importer/src/path_canonicalizer.rs

```diff
@@ -219,17 +219,67 @@ fn canonicalize_inside_expression(
     paths: &'_ PathMap,
 ) {
     e.pre_visit_expressions_mut(&mut |e| {
-        if let Expression::Reference(reference) = e {
-            // If resolving the reference fails, we assume it is a local variable that has been checked below.
-            if let Some(n) = paths.get(&path.clone().join(reference.path.clone())) {
-                *reference = n.relative_to(&Default::default()).into();
-            } else {
-                assert!(reference.path.try_to_identifier().is_some());
+        match e {
+            Expression::Reference(reference) => {
+                // If resolving the reference fails, we assume it is a local variable that has been checked below.
+                if let Some(n) = paths.get(&path.clone().join(reference.path.clone())) {
+                    *reference = n.relative_to(&Default::default()).into();
+                } else {
+                    assert!(reference.path.try_to_identifier().is_some());
+                }
+            }
+            Expression::BlockExpression(statements, _expr) => {
+                for statement in statements {
+                    if let StatementInsideBlock::LetStatement(let_statement) = statement {
+                        canonicalize_inside_pattern(&mut let_statement.pattern, path, paths);
+                    }
+                }
             }
+            Expression::LambdaExpression(lambda) => {
+                lambda.params.iter_mut().for_each(|p| {
+                    canonicalize_inside_pattern(p, path, paths);
+                });
+            }
+            Expression::MatchExpression(_, match_arms) => {
+                match_arms.iter_mut().for_each(|MatchArm { pattern, .. }| {
+                    canonicalize_inside_pattern(pattern, path, paths);
+                })
+            }
+            _ => {}
         }
     });
 }
 
+fn canonicalize_inside_pattern(
+    pattern: &mut Pattern,
+    path: &AbsoluteSymbolPath,
+    paths: &'_ PathMap,
+) {
+    match pattern {
+        Pattern::Enum(name, None) => {
+            // TODO Currently, we treat any identifier as a variable pattern.
+            // If done properly, single identifiers that resolve to enum values
+            // (like a single None) should not be treated as variables.
+            // This is planned to be fixed with the refactoring of path_canonicalizer,
+            // where we remove the two-step approach.
+            if let Some(name) = name.try_to_identifier() {
```

Comment:
ahh ok, so currently "lone identifier looking" enum values can't be matched because they create a new variable